File: src/jk_5/nailed/map/gameloop/instructions/InstructionSleep.java
package jk_5.nailed.map.gameloop.instructions;

import jk_5.nailed.map.gameloop.GameThread;
import jk_5.nailed.map.gameloop.IInstruction;

/**
 * TODO: Edit description
 *
 * @author jk-5
 */
public class InstructionSleep implements IInstruction {

    private long delay;

    @Override
    public void injectArguments(String arguments) {
        this.delay = Long.parseLong(arguments);
    }

    @Override
    public void execute(GameThread controller) throws InterruptedException {
        Thread.sleep(this.delay);
    }
}

File: src/main/scala/jk_5/nailed/map/gameloop/instructions/InstructionSleep.scala
package jk_5.nailed.map.gameloop.instructions

import jk_5.nailed.map.gameloop.GameThread
import jk_5.nailed.map.gameloop.IInstruction

/**
 * No description given
 *
 * @author jk-5
 */
class InstructionSleep extends IInstruction {
  private var delay: Long = 0L
  def injectArguments(args: String) = this.delay = args.toLong
  def execute(controller: GameThread) = Thread.sleep(this.delay)
}
